arcgis-ios-sdk-samples/Content Display Logic/Cells/ContentTableCell.swift

```diff
@@ -16,46 +16,21 @@ import UIKit
 
 class ContentTableCell: UITableViewCell {
 
+    @IBOutlet var stackView: UIStackView!
     @IBOutlet var titleLabel: UILabel!
     @IBOutlet var detailLabel: UILabel!
-    @IBOutlet var infoButton: UIButton!
-    @IBOutlet var parentView: UIView!
     
-    override func awakeFromNib() {
-        super.awakeFromNib()
-        // Initialization code
-        
-        parentView.layer.borderColor = UIColor(white: 0.85, alpha: 1).cgColor
-        parentView.layer.borderWidth = 1
-    }
-
-    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
-        
-        if highlighted {
-            self.parentView.backgroundColor = .secondaryBlue
-            self.titleLabel.textColor = .white
-            self.detailLabel.textColor = .white
-            self.infoButton.tintColor = .white
-        } else {
-            self.parentView.backgroundColor = .white
-            self.titleLabel.textColor = .primaryTextColor
-            self.detailLabel.textColor = .secondaryTextColor
-            self.infoButton.tintColor = .secondaryBlue
+    var isExpanded: Bool = false {
```

Comment:
Oh you're right. Interesting, typically when you hide a view it still participates in the layout.

Reply:
We're currently getting the desired animation behavior so I don't think we need a way to manually flag animations right now.

Reply:
> We're currently getting the desired animation behavior

Ah, I see. You're animating the expansion by reloading the cell. That works, but it has a couple drawbacks:

1. It causes the cell to be reloaded. That doesn't really change anything other than updating the `isExpanded` state of the cell. It would be more efficient to set that directly.
2. The animation is a bit strange. The detail button disappears and reappears in a different spot rather than moving to the new location.

I played around with animating the expansion rather than the reloading of the cell. Forget what I said about having the cell perform that animation; that doesn't work. There's no way that I know of for the cell to trigger a resize of itself inside the table view. This almost works:

```swift
if cell.isExpanded {
    UIView.animate(withDuration: 0.2, animations: {
        cell.detailLabel.alpha = 0
    }, completion: { (finished) in
        cell.isExpanded.toggle()
        self.tableView.performBatchUpdates(nil)
    })
} else {
    cell.detailLabel.alpha = 0
    cell.isExpanded.toggle()
    tableView.performBatchUpdates(nil) { finished in
        guard finished else { return }
        UIView.animate(withDuration: 0.2, animations: {
            cell.detailLabel.alpha = 1
        })
    }
}
```

The only remaining problem is that on collapse, the `titleLabel`'s frame is immediately expanded to fill the whole stack view. And since a label's text is vertically centered, the title jumps and then animates back to where it should have been all along. That can be fixed by subclassing `UILabel` to vertically align the text to the top, but that's probably an exercise for another day. Reloading the cell works well enough for now.

Reply:
Thanks for looking into this. It also struck me as awkward to reload the cells in order to expand them (that was the existing behavior, I think). You're also right about the odd detail button animation. I didn't look into it further since it's still better than it was and we can change it later. It's also not in actual sample code so it's unlikely developers will be studying it.